src/os/bluestore/AvlAllocator.cc

```diff
@@ -342,10 +395,13 @@ AvlAllocator::AvlAllocator(CephContext* cct,
     cct->_conf.get_val<uint64_t>("bluestore_avl_alloc_bf_threshold")),
   range_size_alloc_free_pct(
     cct->_conf.get_val<uint64_t>("bluestore_avl_alloc_bf_free_pct")),
-  max_search_count(
-    cct->_conf.get_val<uint64_t>("bluestore_avl_alloc_ff_max_search_count")),
-  max_search_bytes(
-    cct->_conf.get_val<Option::size_t>("bluestore_avl_alloc_ff_max_search_bytes")),
+  nf_search_time(
```

Comment:
If we stick with the legacy config parameter access (ie cct->_conf->bluestore_avl_alloc_nf_search_time) we can track their modifications with a zero overhead in real-time rather then exclusively on startup. Hence one would be able to tune them without OSD restart...

Reply:
yeah, that's a good point.  I think we can do it with the new style with some trickery right?

Reply:
FWIW I *like* using the legacy interface and hate how fragmented it all has become, but I feel a little guilty for keeping using it. :)  Having said that, I'm not at all opposed to using it especially in the hot path if other people feel the same way.